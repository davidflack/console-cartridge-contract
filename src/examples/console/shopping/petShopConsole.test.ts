import { mockPetShopCartridge } from './__fixtures__';
import { getPetShopConsole, listInventory } from './petShopConsole';

describe(listInventory, () => {
  it('loads a pet shop cartridge and calls the correct operation', async () => {
    const list = listInventory(mockPetShopCartridge);
    expect(await list()).toMatchSnapshot();
  });
});

describe(getPetShopConsole, () => {
  it('loads the correct cartridge', () => {
    expect(getPetShopConsole(mockPetShopCartridge)).toMatchSnapshot();
  });
});
