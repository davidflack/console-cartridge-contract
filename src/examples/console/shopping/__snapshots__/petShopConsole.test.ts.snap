// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPetShopConsole loads the correct cartridge 1`] = `
Object {
  "listInventory": [Function],
}
`;

exports[`listInventory loads a pet shop cartridge and calls the correct operation 1`] = `
Object {
  "description": "Dog Chew Toy",
  "name": "Dog Toy",
  "price": 10,
  "sku": "PET-BB-PUR-01",
}
`;
